docs/develop/java/overview.mdx

```diff
@@ -4,79 +4,100 @@ description: "Get an idea of what a Restate Java service looks like."
+Run:
 
-Below is an example of a Java service. 
+```shell
+gradle init --type kotlin-application
+```
 
-Restate uses [protobuf](https://protobuf.dev/) to define the service and its methods. The example shows a greeter service with a single method:
-- `greet`: maintains a record of the number of times it has received a request for a given name, and prints a greeting.
+Add the [Kotlin symbol processing](https://kotlinlang.org/docs/ksp-quickstart.html#use-your-own-processor-in-a-project) and the [Kotlin serialization](https://kotlinlang.org/docs/serialization.html) plugin:
 
-```java
-CODE_LOAD::java/src/main/java/develop/Greeter.java
+```kotlin
+plugins {
+    kotlin("plugin.serialization") version "1.9.22"
+    id("com.google.devtools.ksp") version "1.9.22-1.0.18"
+}
 ```
 
-The `Greeter` class extends `GreeterRestate.GreeterRestateImplBase`, which is the base generated by the Restate protoc generator.
-It has the name of the service (`Greeter`) with `Restate` appended to it.
+Add the runtime dependency `sdk-api-kotlin` and the ksp dependency `sdk-api-kotlin-gen`:
 
-The [Restate context](/develop/java/restate-context) is supplied to the method as the first argument.
-The context enables interaction with Restate (e.g. calling other services, setting timers).
-The `greet` method uses an ObjectContext since it is a keyed service. For unkeyed services, use `Context`.
-Finally, the Restate server is set up to serve both methods of the greeter service.
+```kotlin
+ksp("dev.restate:sdk-api-kotlin-gen:VAR::JAVA_SDK_VERSION")
+implementation("dev.restate:sdk-api-kotlin:VAR::JAVA_SDK_VERSION")
+```
 
-This is what corresponding Protobuf definition would look like for this service:
+</TabItem>
+</Tabs>
 
-```protobuf
-CODE_LOAD::java/src/main/proto/greeter.proto
-```
+Depending on the deployment target, add one of the following dependencies:
+
+- To run as long-running app: `dev.restate:sdk-http-vertx:VAR::JAVA_SDK_VERSION`
+- To run on AWS Lambda: `dev.restate:sdk-lambda:VAR::JAVA_SDK_VERSION`
 
-To understand the Restate-specific parts of this Protobuf definition, have a look at the documentation [here](/develop/service_type#restate-service-contract).
+## API Examples
 
-<details>
-<summary>Alternative: vanilla gRPC</summary>
+Below is an example of a virtual object.
 
-Restate's proto code generator makes the generated code more polished for writing Restate services.
-However, it is also possible to use the vanilla generated protobuf code by `protoc-gen-grpc-java`, in case you prefer this.
+The Java/Kotlin SDK uses annotations to define the component and its handlers.
+The example shows a greeter service with a single handler `greet`, that maintains a record of the number of times it has received a request for a given name, and prints a greeting.
 
-An example:
+<Tabs groupId="languages">
+<TabItem value="java" label="Java">
 
 ```java
-CODE_LOAD::java/src/main/java/develop/GreeterVanillaGrpc.java
+CODE_LOAD::java/src/main/java/develop/Greeter.java
```

Comment:
Can we update this example to be similar to the template? So with the greeting as the argument and the name as the virtual object key? Also for Kotlin.